File: Phase01-Search/src/test/java/model/SearchQueryTest.java
package model;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

public class SearchQueryTest {
    @Test
    void SearchQueryConstructorTest() {
        String firstInput = "+fastest +holding -easily -stresses need information";
        String[] firstRequiredWordsExpected = {"need", "inform"};
        String[] firstOptionalWordsExpected = {"hold", "fastest"};
        String[] firstBannedWordsExpected = {"stress", "easili"};
        SearchQuery firstSearchQuery = new SearchQuery(firstInput);
        Assertions.assertAll(() -> {
            Assertions.assertArrayEquals(firstRequiredWordsExpected, firstSearchQuery.getRequiredWords());
            Assertions.assertArrayEquals(firstOptionalWordsExpected, firstSearchQuery.getOptionalWords());
            Assertions.assertArrayEquals(firstBannedWordsExpected, firstSearchQuery.getBannedWords());
        });
    }
}

File: Phase08-EFCore/SearchLib/SearchLib.Test/Model/SearchQueryTest.cs
using FluentAssertions;
using SearchLib.Model;
using Xunit;

namespace SearchLib.Test.Model
{
    public class SearchQueryTest
    {
        [Fact]
        void SearchQueryConstructorTest()
        {
            string[] requiredWordsExpected = { "need", "inform" };
            string[] optionalWordsExpected = { "hold", "fastest" };
            string[] bannedWordsExpected = { "stress", "easili" };
            string input = "+fastest +holding -easily -stresses need information";
            SearchQuery searchQuery = new SearchQuery(input);
            searchQuery.RequiredWords.Should().BeEquivalentTo(requiredWordsExpected);
            searchQuery.OptionalWords.Should().BeEquivalentTo(optionalWordsExpected);
            searchQuery.BannedWords.Should().BeEquivalentTo(bannedWordsExpected);
        }
    }
}
